src/services/impl/FileTextReader.hh

```diff
@@ -0,0 +1,52 @@
+#ifndef RA_SERVICES_FILETEXTREADER
+#define RA_SERVICES_FILETEXTREADER
+#pragma once
+
+#include "services\TextReader.hh"
+
+#include "RA_StringUtils.h"
+
+#include <fstream>
+
+namespace ra {
+namespace services {
+namespace impl {
+
+class FileTextReader : public ra::services::TextReader
+{
+public:
+    FileTextReader(const std::wstring& sFilename)
+        : m_iStream(sFilename)
+    {
+    }
+
+    bool GetLine(_Out_ std::string& sLine) override
+    {
+        if (!std::getline(m_iStream, sLine))
+            return false;
+
+        ra::TrimLineEnding(sLine);
+        return true;
+    }
+
+    bool GetLine(_Out_ std::wstring& sLine) override
+    {
+        std::string sNarrowLine;
+        if (!std::getline(m_iStream, sNarrowLine))
+            return false;
+
+        sLine = ra::Widen(ra::TrimLineEnding(sNarrowLine));
```

Comment:
Not a big deal but you don't have to use the fully qualified name inside the same namespace but if that's what you prefer that's OK.